### PR TITLE
Add .csv download button to data explorer

### PR DESCRIPTION
**Pull request by ebb-tide**
- [x] CHANGELOG.md updated with a link to the PR (not the Issue)
  - [x] Rebased/mergable
  - [x] Tests pass
  - [x] Sign [CLA](https://influxdata.com/community/cla/) (if not already signed)

Connect [#1405](https://github.com/influxdata/chronograf/issues/1405)

downloadcsv

### The problem
Users would like to be able to download their data from the data explorer in .csv format. 

### The Solution
Users are able to download their data in .csv format. The .csv download button calls an action to receive the most updated data available.

### CODE REVIEW

File: ui/src/shared/parsing/resultsToCSV.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+          ','
+        )
+      )
+    )
+    .join('\n')
+
+  return {name, CSVString}
+}
+
+export default resultsToCSV
```
Comment (121watts) [inline ui/src/shared/parsing/resultsToCSV.js]: Please test this function :)

File: ui/src/data_explorer/components/Visualization.js [hunk 2]
```diff
@@ -90,13 +95,13 @@ class Visualization extends Component {
...
             axes={axes}
+            query={query}
             queries={queries}
             templates={templates}
             cellType={cellType}
             autoRefresh={autoRefresh}
             heightPixels={heightPixels}
             editQueryStatus={editQueryStatus}
-            activeQueryIndex={activeQueryIndex}
             isInDataExplorer={isInDataExplorer}
```
Comment (121watts) [inline ui/src/data_explorer/components/Visualization.js]: +1 to removing the need for `activeQueryIndex`

File: ui/src/data_explorer/components/VisHeader.js [hunk 3]
```diff
@@ -2,34 +2,55 @@ import React, {PropTypes} from 'react'
...
+import download from 'src/external/download.js'
+
+const getCSV = query => async () => {
+  try {
+    const {results} = await fetchTimeSeriesAsync({source: query.host, query})
+    const {name, CSVString} = resultsToCSV(results)
+    download(CSVString, `${name}.csv`, 'text/plain')
+  } catch (error) {
+    console.error(error)
+  }
```
Comment (121watts) [inline ui/src/data_explorer/components/VisHeader.js]: Let's also publish and error to the user here to let them know something went wrong

**Comment by jaredscheib** [comment]
Nice work @ebb-tide! Exciting to see this PR! :)

I've only looked at the code, but could you double-check that the license CCBY2.0 is acceptable for use in `master`? @timhallinflux can advise on that.

**Comment by 121watts** [comment]
LGTM!

**Comment by mkarlesky** [comment]
@ebb-tide It was my community post that spawned [#1405](https://github.com/influxdata/chronograf/issues/1405). Just wanted to thank you for your work on this! So looking forward to having it available.

**Pull request merged**